File: frontend/src/api/runtime-status.ts
import { apiGet } from './client';

// GET /api/runtime-status(04A §2.1 运行态面板 / 诊断中心,基础设施只读投影)。
// 与 health.ts 同理:infra 端点手写类型,不走 generated schema。

export interface RuntimeComponentRead {
  key: string; // api | db | redis | worker | event_bus
  label: string;
  status: 'ok' | 'degraded' | 'down' | 'not_applicable' | string;
  detail?: string | null;
}

export interface RuntimeAlertRead {
  code: string;
  level: 'WARN' | 'ERROR' | string;
  summary: string;
  hint?: string | null;
}

export interface AsyncJobsSummaryRead {
  mode: 'inline' | 'queued' | string;
  queued: number | null;
  running: number | null;
  failed_recent: number | null;
  oldest_waiting_minutes: number | null;
  queue_depth: number | null;
}

export interface RecentAgentRunRead {
  run_id: string;
  kind: string; // 稳定码
  kind_label: string; // 白话名(后端单一来源,前端不再自建映射)
  status: 'queued' | 'started' | 'succeeded' | 'failed' | string;
  created_at: string; // 发起时刻 ISO
  duration_seconds: number | null; // 终态才有;非终态为 null
  reason_code: string | null; // 失败行的原因稳定码(不含 error 原文)
}

export interface DiagnosticEventRead {
  event: string;
  component: string;
  level: string;
  first_seen: string;
  last_seen: string;
  count: number;
}

export interface RuntimeStatusRead {
  status: 'normal' | 'degraded' | 'down' | string;
  alert_count: number;
  generated_at: string;
  components: RuntimeComponentRead[];
  alerts: RuntimeAlertRead[];
  async_jobs: AsyncJobsSummaryRead;
  recent_jobs?: RecentAgentRunRead[]; // 加性扩展:旧后端不返回该字段
  diagnostics: DiagnosticEventRead[];
}

export const runtimeStatusApi = {
  getRuntimeStatus(): Promise<RuntimeStatusRead> {
    return apiGet<RuntimeStatusRead>('/runtime-status');
  },
};

File: backend/app/api/schemas/runtime_status.py
"""运行态面板与诊断中心（基础设施只读投影）。

自 schemas.py 按业务域拆包（命名规范见包入口 __init__.py）。
"""
from __future__ import annotations

from pydantic import BaseModel


# ---- 运行态面板 / 诊断中心（04A §2.1,基础设施只读投影,非业务事实源）----


class RuntimeComponentRead(BaseModel):
    """单个平台组件的探测结果。status:ok/degraded/down/not_applicable。"""

    key: str  # api | db | redis | worker | event_bus
    label: str
    status: str
    detail: str | None = None


class RuntimeAlertRead(BaseModel):
    """活跃风险组(按组去重,现算不累加)。level:WARN/ERROR。"""

    code: str
    level: str
    summary: str
    hint: str | None = None


class AsyncJobsSummaryRead(BaseModel):
    """异步作业摘要(agent_run 表聚合 + Redis 队列深度)。"""

    mode: str  # inline | queued
    queued: int | None = None
    running: int | None = None
    failed_recent: int | None = None  # 近 24h 失败数
    oldest_waiting_minutes: int | None = None
    queue_depth: int | None = None  # Redis 队列积压(inline 模式为 None)


class RecentAgentRunRead(BaseModel):
    """最近一条异步作业(运行态面板明细表一行)。

    只投影稳定码与派生读数:kind_label 是给人看的白话名,reason_code 是登记过的失败原因码,
    AgentRun.error 的原文一律不出现在本结构里(硬规则 8)。
    """

    run_id: str
    kind: str  # AgentRun.kind 稳定码
    kind_label: str  # 白话名(单一来源=services/notification.AGENT_RUN_KIND_LABELS)
    status: str  # queued | started | succeeded | failed
    created_at: str  # 发起时刻 ISO
    duration_seconds: int | None = None  # 终态才有;非终态=None(等待中/进行中)
    reason_code: str | None = None  # 失败行的原因稳定码


class DiagnosticEventRead(BaseModel):
    """诊断事件白名单摘要(进程内环形缓冲,只读展示)。"""

    event: str
    component: str
    level: str
    first_seen: str
    last_seen: str
    count: int


class RuntimeStatusRead(BaseModel):
    """GET /api/runtime-status。status:normal/degraded/down;alert_count=活跃风险组数。"""

    status: str
    alert_count: int
    generated_at: str
    components: list[RuntimeComponentRead]
    alerts: list[RuntimeAlertRead]
    async_jobs: AsyncJobsSummaryRead
    recent_jobs: list[RecentAgentRunRead] = []  # 最近异步作业明细(加性扩展)
    diagnostics: list[DiagnosticEventRead]
